### DISCUSSION TITLE
Is there any way to get the search query value in another component

### DISCUSSION THREAD

**Discussion by EmmanuelVictor62** [algolia/instantsearch#6255] [Q&A] [answered]
Hi Algolia community,

I have been working on a react application in which algolia is integrated, I have an AlgoliaSearchInput component which I am importing into the search modal component, I want to get the search query so I can use this to filter by facets and render the hits, problem is when I try the conventional of doing this by lifting the search query state, it cause the app to freeze, and when I try to pass down state from the parent component to the input box, there is no way to call the refine method on Algolia, can you suggest a solution to this, here is a sample code of what I am currently implementing

![image](https://github.com/algolia/instantsearch/assets/104584134/96970325-d889-4546-8524-def172a9d8e6)
![image](https://github.com/algolia/instantsearch/assets/104584134/11c083cb-c096-4422-ba59-8ba6193dad59)

**Comment by Haroenv** [comment]
Do you have a reproduction? I can't see what's wrong with the code you provided here. Maybe also look into `useInstantSearch().setUiState` if you want to set a query and filters at the same time

**Reply by EmmanuelVictor62** [reply]
This is the problem @Haroenv, I am not getting any error, no error in my console, just that the app freezes after a while and crashes my code editor, the problem gets resolved once I remove the code and btwn I am using react-instantsearch-dom and the useInstantSearch module is not available on it

**Reply by Haroenv** [reply]
ah I see, you're using v6? please update to v7 first, this is indeed a known issue related to nested HoC causing infinite loops which is fixed in v7

**Reply by EmmanuelVictor62** [reply]
Yes @Haroenv, I also noticed that the issue is caused by endless loops which eventually crash the app, I will update now and share the feedback with you after implementing this

**Reply by EmmanuelVictor62** [reply]
@Haroenv, I have the latest version of react-instantsearch-dom installed `"react-instantsearch-dom": "^6.40.4",` and algolia search `"algoliasearch": "^4.24.0"` but the issue still persists, do you think I should migrate to react-instantsearch rather than the instantsearch-dom??

**Reply by Haroenv** [reply, marked as answer]
the package is indeed called `react-instantsearch`, not `react-instantsearch-dom` which has been deprecated since July last year. You should indeed upgrade to `react-instantsearch@7`

**Reply by EmmanuelVictor62** [reply]
I have upgraded now and everything works fine, thanks for the support @Haroenv
